utkdigitalinitiatives/basic-solr-config
# 4.x - Solr search exclusions and Fedora Generic Search multi-threading

# What does this pull request do?

This PR provides the following updates:
1. Solr search exclusions (e.g. undescribed collection and islandora namespace objects)
2. Configuration files for multi-threading the Fedora Generic Search (FGS) webapp
# What's new?

Lots. There are updates to the README, new directories for the various configuration files,  and modifications to those configuration files.
# How should this be tested?

The terrible truth is that some of these changes have already be instantiated on dev and production; this is the case for the FGS performance tweaks. Testing/reviewing the changes made to solr-conf/solrconfig.xml is possible:
_note: this assumes that you are using a minimal vm and do not have islandora_collection_search installed._
- start your vm
- visit http://localhost:8080/solr/collection1/select?q=_%3A_&fl=PID&wt=xml&indent=true
  - this search returns the first 10 items in the index, and should look something like this:

``` xml
<response>

<lst name="responseHeader">
  <int name="status">0</int>
  <int name="QTime">0</int>
  <lst name="params">
    <str name="q">*:*</str>
    <str name="indent">true</str>
    <str name="fl">PID</str>
    <str name="wt">xml</str>
  </lst>
</lst>
<result name="response" numFound="46" start="0">
  <doc>
    <str name="PID">islandora:bookCollection</str></doc>
  <doc>
    <str name="PID">islandora:compound_collection</str></doc>
  <doc>
    <str name="PID">islandora:sp_disk_image_collection</str></doc>
  <doc>
    <str name="PID">islandora:sp_pdf_collection</str></doc>
  <doc>
    <str name="PID">islandora:video_collection</str></doc>
  <doc>
    <str name="PID">islandora:sp_web_archive_collection</str></doc>
...[snip]...
```

```
- this example will exclude the first two results, `islandora:bookCollection` and `islandora:compound_collection`
```
- use your favorite text editor to modify /usr/local/solr/collection1/conf/solrconfig.xml
- at around line 800, add the following and save the file

``` xml
<lst name="appends">
  <str name="fq">-PID:islandora\:bookCollection -PID:islandora\:compound_collection</str>
</lst>
```
- restart Tomcat; i.e. sudo service tomcat7 restart
- after Tomcat has restarted, revisit the previous link (http://localhost:8080/solr/collection1/select?q=_%3A_&fl=PID&wt=xml&indent=true) and the PIDs you chose to exclude should not be in the results.
